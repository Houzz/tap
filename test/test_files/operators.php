<?php

function f() {
    $x = true ? 1 : "str";
    __tap_check($x, "int|string");
    $x = "str" ? "yes" : "no";
    $x = (1 << 2);
    __tap_check($x, "int");
}

function f2() {
    $x = 1;
    $x += 2;

    $y = 1;
    $y <<= 2;
}

function f3() {
    $x = (1 <= 2);
    __tap_check($x, "bool");
    $x = ("str" <= 2);  // error
    $x = (1 == 3);
    $x = (1 == "str");  // error
}

