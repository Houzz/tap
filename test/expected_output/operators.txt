6 E_EXPECTED_TYPE
22 E_EXPECTED_TYPE
24 E_EXPECTED_TYPE
